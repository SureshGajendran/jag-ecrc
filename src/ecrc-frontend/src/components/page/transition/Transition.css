.transition-content {
  flex-direction: column;
  padding-top: 30px;
  padding-bottom: 270px;
}
