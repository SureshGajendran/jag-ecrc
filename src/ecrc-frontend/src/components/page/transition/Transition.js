/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";
import PropTypes from "prop-types";
import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import "./Transition.css";

export default function Transition({ page: { header, transitionReason } }) {
  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="transition-content col-md-12">
          <p>
            You are being redirected to the legacy online criminal record check
            (eCRC) application. You will be required to re-enter your access
            code to continue. If you are not redirected automatically, please
            click <a href="https://justice.gov.bc.ca/eCRC/home.htm">here</a>
            {"."}
          </p>
          <br />
          {transitionReason === "notwhitelisted" && (
            <p>
              We are transitioning our client organizations currently using
              Equifax for identity verification to a new process through a{" "}
              <a href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card">
                BC Services Card account
              </a>
              {". Your organization hasn't"} been transitioned to the new system
              yet.
            </p>
          )}
          {transitionReason === "bcsc" && (
            <p>
              TEXT TO BE UPDATED You failed a bcsc login, did not have a bcsc
              with picture, did not have bcsc...etc
            </p>
          )}
        </div>
      </div>
      <Footer isSmallPage />
    </main>
  );
}

Transition.propTypes = {
  page: PropTypes.shape({
    header: PropTypes.shape({
      name: PropTypes.string.isRequired
    }).isRequired,
    transitionReason: PropTypes.string
  })
};

Transition.defaultProps = {
  page: {
    transitionReason: "bcsc"
  }
};
