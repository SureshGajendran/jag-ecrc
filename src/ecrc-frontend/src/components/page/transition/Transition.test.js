import React from "react";
import { create } from "react-test-renderer";

import Transition from "./Transition";

describe("Transition Page Component", () => {
  test("Matches the snapshot", () => {
    const header = {
      name: "Criminal Record Check"
    };

    const transition = create(<Transition header={header} />);
    expect(transition.toJSON()).toMatchSnapshot();
  });
});
