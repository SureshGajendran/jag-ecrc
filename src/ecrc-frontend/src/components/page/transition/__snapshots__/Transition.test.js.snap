// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transition Page Component Matches the snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="https://www2.gov.bc.ca"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div
          className="navbar-brand"
        >
          Criminal Record Check
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav"
          >
            <a
              className="nav-item nav-link"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="transition-content col-md-12"
    >
      <p>
        You are being redirected to the legacy online criminal record check (eCRC) application. You will be required to re-enter your access code to continue. If you are not redirected automatically, please click 
        <a
          href="https://justice.gov.bc.ca/eCRC/home.htm"
        >
          here
        </a>
        .
      </p>
      <br />
      <p>
        We are transitioning our client organizations currently using Equifax for identity verification to a new process through a
         
        <a
          href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card"
        >
          BC Services Card account
        </a>
        . Your organization hasn't
         been transitioned to the new system yet.
      </p>
    </div>
  </div>
  <footer
    className="footer"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
